streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/trait/FlinkClientTrait.scala

```diff
@@ -156,6 +156,27 @@ trait FlinkClientTrait extends Logger {
           })
     }
 
+    // set prometheus
+    val enablePrometheus: Boolean =
+      InternalConfigHolder.get(CommonConfig.STREAMPARK_PROMETHEUS_ENABLE)
+    val promgatewayHostUrl: String =
+      InternalConfigHolder.get(CommonConfig.STREAMPARK_PROMGATEWAY_HOST_URL)
+
+    if (
+      submitRequest.executionMode == FlinkExecutionMode.YARN_APPLICATION && enablePrometheus && StringUtils
```

Comment:
Why is it only enabled in yarn application mode? Doesn't it work in other modes?

Reply:
At that time, only yarn application mode was tested